packages/react-compose/src/next/getSlots.test.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { getSlots } from './getSlots';
+import { nullRender } from './nullRender';
+
+describe('getSlots', () => {
+  const Foo = () => <div />;
+
+  it('renders nullRender for root if the as prop is not provided', () => {
+    expect(getSlots({})).toEqual({
+      slots: { root: nullRender },
+      slotProps: { root: {} },
+    });
+  });
+
+  it('renders root as a span with no props', () => {
+    expect(getSlots({ as: 'span' })).toEqual({
+      slots: { root: 'span' },
+      slotProps: { root: {} },
+    });
+  });
+
+  it('renders root as a button, omitting the href prop', () => {
```

Comment:
```suggestion
  it('omits props which are not valid for the element rendered', () => {
```
I think this better explain what we are trying to test

Reply:
much better :)